src/libostree/ostree-repo.c

```diff
@@ -2094,6 +2094,27 @@ append_remotes_d (OstreeRepo          *self,
 }
 
 gboolean
+ostree_repo_exists (OstreeRepo    *self,
+                    gboolean      *does_exist,
+                    GError        **error)
+{
+  struct stat stbuf;
+
+  if (lstat (gs_file_get_path_cached (self->objects_dir), &stbuf) < 0)
```

Comment:
Hmm, wouldn't it be a stronger test if we actually checked that it was a valid repo? E.g. call `ostree_repo_open()` on a separate reference and check for errors? Though I guess then, the function would be more appropriately called something like `ostree_repo_is_valid()`.